astropy/cosmology/flrw.py

```diff
@@ -1485,6 +1448,56 @@ def arcsec_per_kpc_proper(self, z):
         return u.arcsec / (self.angular_diameter_distance(z).to(u.kpc) *
                            arcsec_in_radians)
 
+    # -----------------------------------------------------
+
+    def equivalent(self, other):
+        # Note! this method is superseded by FlatFLRWMixin.equivalent in
+        # subclassses of FlatFLRWMixin.
+
+        # case (3) iff FlatFLRWMixin, so delegate check to FlatFLRWMixin
+        if isinstance(other, FlatFLRWMixin):
+            return other.equivalent(self)
+
+        # can now pass to super, to check if the class & params are the same
+        return super().equivalent(other)
+
+
+class FlatFLRWMixin(FlatCosmologyMixin):
+
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+        # Do some twiddling after the fact to get flatness
+        self._Ode0 = 1.0 - self._Om0 - self._Ogamma0 - self._Onu0
+        self._Ok0 = 0.0
+
+    @property  # no longer a Parameter
+    def Ode0(self):
+        """Omega dark energy; dark energy density/critical density at z=0"""
+        return self._Ode0
+
+    def equivalent(self, other):
+        # check if case (1): same class & parameters
+        if isinstance(other, FlatFLRWMixin):
+            return Cosmology.equivalent(self, other)
```

Comment:
That makes it feel wrong. Would giving both classes a chance help? I.e., have a general `equivalent()` method which calls private `_equivalent()` methods, first on `self`, and if `NotImplemented`, on `other`?

Reply:
I'm coming around. Still looking into this.

Reply:
Done! it works great. Thanks.